File: src/main/java/net/hectus/neobb/turn/default_game/throwable/ThrowableTurn.java
package net.hectus.neobb.turn.default_game.throwable;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.util.Utilities;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.entity.Projectile;
import org.bukkit.inventory.ItemStack;

public abstract class ThrowableTurn extends Turn<Projectile> {
    public ThrowableTurn(NeoPlayer player) { super(player); }
    public ThrowableTurn(Projectile data, Location impact, NeoPlayer player) { super(data, impact, player); }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.valueOf(Utilities.camelToSnake(Utilities.counterFilterName(getClass().getSimpleName())).toUpperCase()));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/throwable/ThrowableTurn.kt
package net.hectus.neobb.turn.default_game.throwable

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.Turn
import net.hectus.neobb.util.material
import org.bukkit.entity.Projectile
import org.bukkit.inventory.ItemStack

abstract class ThrowableTurn(data: Projectile?, cord: Cord?, player: NeoPlayer?) : Turn<Projectile>(data, cord, player) {
    override fun item(): ItemStack = ItemStack(this::class.material())
}
